cmd/erasure-healfile.go

```diff
@@ -16,72 +16,136 @@
 
 package cmd
 
-import "hash"
+import (
+	"hash"
+)
+
+// HealFile tries to reconstruct a bitrot encoded file spread over all
+// available disks. HealFile will read the valid parts of the file,
+// reconstruct the missing data and write the reconstructed parts back
+// to `staleDisks`.
+//
+// `staleDisks` is a slice of disks where each non-nil entry
+// represents an online disk that is known to have stale or no data,
+// and so will be healed.
```

Comment:
Maybe add:
> `staleDisks` are used to avoid writes to unavailable dsiks and support partial healing

This was my idea of HealFile - I thought we try to heal all disks which are marked as offline and fail if at least on write fails -> no partial heal. Now I see that heal tries to heal a set of disks which are explicitly marked as "healable". However, we still fail if one of the "healable" disks goes down.

Reply:
Partial healing is supported in the sense that disks that are offline may be having out-of-date data, but we still continue and heal online disks with stale data. The comment as it stands, IMO  expresses what the function is doing correctly, and what your comment says is already implicitly implied.

The point about failing if one of the stale disks goes down while writing is well-taken. It can occasionally happen, though the caller of HealFile already confirms that they are online. i will look at enhancing this in a subsequent patch to return errors per-disk in case of write failures, and also optimizing the calling code that appears to do multiple passes over the data - many of these optimizations are already present in the new admin API PR. However, it would be better to separate out those optimizations into a new patch, and reduce the amount of low level changes in the admin API PR.